checker.js

```diff
@@ -1,17 +1,50 @@
 const expect = require('chai').expect;
-
-function vitalsAreOk(bpm, spo2, respRate) {
-    if(bpm < 70 || bpm > 150) {
-        return false;
-    } else if(spo2 < 90) {
-        return false;
-    } else if(respRate < 30 || respRate > 95) {
-        return false;
+function ReportGenerator(){
+    this.checkBpm = function(bpm){
+        if(bpm < 70){
```

Comment:
This function is very similar to the other check functions. In JavaScript, it is possible to make a function that returns another function. So you can make a function that returns range-checking functions with different limits for different vitals. Try it out and see, it would be a good experience.

Reply:
I think i have done that already. The lines you have given me comments for, are no more in my code. In vitalsAreOk() i am now returning multiple functions each checking the range for a particular vital. Can you verify it please?

Reply:
No I was referring to the new code itself. The this.check... variables are assigned to functions that are very similar. Instead of coding each of them (and repeating for new vitals), you can use closure and make one function that returns those functions.

Reply:
Look at this for ideas...
https://stackoverflow.com/questions/7629891/functions-that-return-a-function

Reply:
I did some changes. Would like you to review it. I also looked at other's solution for this question but could not find anything much different. Suggestions?

Reply:
Instead of different check... functions, make a function that can make other check functions. Try this:
```JavaScript
function makeChecker(low, high) {
  function isInRange(value) {
    return value >= low && value <= high;
  }
  return isInRange;
}
```
And in the caller:
```JavaScript
bpmIsInRange = makeChecker(70, 150);
if(!bpmIsInRange(160)) {
  //...
}
```

Reply:
Yes Sudeep, i have done changes as per your suggestions. Tried following SOLID principles too. The code seems much better now. Thank you for the suggestions.